src/contexts/CurrencyContext/CurrencyContext.tsx

```diff
@@ -0,0 +1,94 @@
+import { FC, createContext, useEffect, useState, PropsWithChildren } from "react";
+import { httpClient } from "../../common";
+import { AxiosResponse } from "axios";
+import { format, subDays } from "date-fns";
+import {
+  FetchedCurrenciesDTO,
+  FetchedCurrenciesHistoryDTO,
+  CurrencyType,
+  FetchedCurrencyNamesType,
+  CurrencyContextType,
+  CurrencyRates,
+  PastCurrencyRates,
+} from "../../types";
+
+export const CurrencyContext = createContext<CurrencyContextType>(null);
+
+export const CurrencyContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [fetchedCurrencyNames, setCurrencyNames] = useState<FetchedCurrencyNamesType>({ currencyCode: "" });
+  const [latestCurrencyRates, setLatestCurrencyRates] = useState<CurrencyRates>({
+    currencyCode: 0,
+  });
+  const [pastCurrencyRates, setPastCurrencyRates] = useState<PastCurrencyRates>({ "": { currencyCode: 0 } });
+  const [presentCurrency, setPresentCurrency] = useState<CurrencyType>(null);
+  const [baseCurrency, setBaseCurrency] = useState("AUD");
+
+  useEffect(() => {
+    httpClient.get(`/latest?from=${baseCurrency}`).then((response: AxiosResponse) => {
+      const fetchedCurrenciesDTO: FetchedCurrenciesDTO = response.data;
+      setLatestCurrencyRates(fetchedDataMapper(fetchedCurrenciesDTO));
+      updatePresentCurrency();
+      function fetchedDataMapper(fetchedCurrencies: FetchedCurrenciesDTO) {

```

Comment:
mapery dobrze jest definiować osobno poza context-em, oraz nadawać nazwy map<Co>To<doCzego>()